public class ExceptionHandling{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try {
            System.out.print("Enter value of a: ");
            int a = scanner.nextInt();
            int b = 0; // Intentionally setting b to 0
            System.out.println("Attempting division: " + a + " / " + b);
            int result = a / b;  
            System.out.println("Result: " + result);
            int[] array = new int[5];
            System.out.println("Accessing the 6th position of the array...");
            int value = array[5];  
            
        } catch (ArithmeticException e) {
            // Catching division by zero
            System.out.println("Error: Division by zero is not allowed.");

        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Error: Array index out of bounds.");

        } finally {
            System.out.println("The 'finally' block executed.");
        }
    }
}
